rust-symcrypt/src/ecdh.rs

```diff
@@ -0,0 +1,257 @@
+/* EcDh functions. For further documentation please refer to symcrypt.h */
+
+use crate::ecurve::*;
+use crate::errors::SymCryptError;
+use std::vec;
+use symcrypt_sys;
+
+/// EcDhKey is a wrapper around symcrypt_sys::PSYMCRYPT_ECKEY, this is to let rust handle the free'ing of this pointer
+/// EcDhKey must be allocated after EcDhCurve and free'd before EcDhCurve is free'd
+pub struct EcDhKey {
+    inner: symcrypt_sys::PSYMCRYPT_ECKEY,
+    curve: EcCurve,
+}
+
+#[derive(Debug)]
+pub struct EcDhSecretAgreement(pub Vec<u8>);
```

Comment:
You may not want to make the `Vec<u8>` field `pub`, since doing so makes it part of your public API and you won't be able to change it without making a breaking change in the future. It's probably better to add something like a `.as_bytes()` method on `EcDhSecretAgreement` which returns `&[u8]` if users do need access to the bytes.